--- ./work/alien-apiservice-1.0.5/C2Perl/C2PERL.c.ori	2005-09-02 15:29:22.000000000 +0200
+++ ./work/alien-apiservice-1.0.5/C2Perl/C2PERL.c	2005-09-02 15:29:41.000000000 +0200
@@ -6,6 +6,8 @@
 #include <stdarg.h>
 #include <string.h>
 
+int g_init;
+
 static PerlInterpreter *my_perl;
 
 static void xs_init (void);
--- ./work/alien-apiservice-1.0.5/C2Perl/C2PERL.h.ori	2005-09-02 15:28:44.000000000 +0200
+++ ./work/alien-apiservice-1.0.5/C2Perl/C2PERL.h	2005-09-02 15:28:56.000000000 +0200
@@ -12,7 +12,7 @@
 #endif
     
     /* the initialization state of the perl module */
-    int g_init;
+extern int g_init;
     
 /**************************************************************************
  * C2PERL API data types.
--- ./work/alien-apiservice-1.0.5/gclientlib/glite_dir_operations.cxx.ori	2005-09-02 15:35:27.000000000 +0200
+++ ./work/alien-apiservice-1.0.5/gclientlib/glite_dir_operations.cxx	2005-09-02 15:40:56.000000000 +0200
@@ -58,7 +58,9 @@
     return 0;
   }
   
+#ifndef __APPLE__
   dir->dir.glite_dir.dir_entry.d_off=0;
+#endif
   int column=0;
   for(column=0; ;column++){
     std::map<std::string, std::string> tags;
@@ -91,16 +93,26 @@
   if(dir->type == sys)
     return readdir(dir->dir.sys_dir);
   struct dirent *dir_entry=&dir->dir.glite_dir.dir_entry;
+#ifndef __APPLE__
   int len = strspn(dir->dir.glite_dir.dirstream+dir_entry->d_off, "\n ");
   dir_entry->d_off+=len;
   len = strcspn(dir->dir.glite_dir.dirstream+dir_entry->d_off, "\n ");
+#else
+  int len = strspn(dir->dir.glite_dir.dirstream, "\n ");
+#endif
   if(!len) 
     return 0;
   if(len>255) len=255;
+#ifndef __APPLE__
   strncpy(dir->dir.glite_dir.dir_entry.d_name,
 	  dir->dir.glite_dir.dirstream +dir_entry->d_off, len);
   dir->dir.glite_dir.dir_entry.d_name[len]='\0';
   dir_entry->d_off += len;
+#else
+  strncpy(dir->dir.glite_dir.dir_entry.d_name,
+	  dir->dir.glite_dir.dirstream, len);
+  dir->dir.glite_dir.dir_entry.d_name[len]='\0';
+#endif
   return dir_entry;
 }
 
@@ -127,8 +139,11 @@
   
   if(dir->type == sys)
     return telldir(dir->dir.sys_dir);
-
+#ifndef __APPLE__
   return dir->dir.glite_dir.dir_entry.d_off;
+#else
+  return 0;
+#endif
 }
 
 
@@ -139,7 +154,9 @@
   if(dir->type == sys)
     rewinddir(dir->dir.sys_dir);
 
+#ifndef __APPLE__
   dir->dir.glite_dir.dir_entry.d_off = 0;
+#endif
 }
 
 
@@ -150,7 +167,9 @@
   if(dir->type == sys)
     seekdir(dir->dir.sys_dir, offset);
 
+#ifndef __APPLE__
   dir->dir.glite_dir.dir_entry.d_off = offset;
+#endif
 }
 
 
--- ./work/alien-apiservice-1.0.5/clients/gsubmit.cxx.ori	2005-09-02 16:09:28.000000000 +0200
+++ ./work/alien-apiservice-1.0.5/clients/gsubmit.cxx	2005-09-02 16:10:26.000000000 +0200
@@ -18,15 +18,15 @@
 
 using namespace std;
 
-string textblack("\033[49;30m");
-string textred("\033[49;31m");
-string textrederror("\033[47;31m");
-string textblueerror("\033[47;31m");
-string textgreen("\033[49;32m");
-string textyellow("\033[49;33m");
-string textblue("\033[49;34m");
-string textbold("\033[1m");
-string textunbold("\033[0m");
+extern string textblack;
+extern string textred;
+extern string textrederror;
+extern string textblueerror;
+extern string textgreen;
+extern string textyellow;
+extern string textblue;
+extern string textbold;
+extern string textunbold;
 
 void print_help() {
   fprintf (stderr, "\n");
