--- work/perl-5.10.1/ext/GDBM_File/Makefile.PL	2009-04-19 19:02:38.000000000 +0200
+++ work/perl-5.10.1/ext/GDBM_File/Makefile.PL.new	2010-04-21 16:49:11.000000000 +0200
@@ -2,7 +2,7 @@
 use ExtUtils::Constant 0.11 'WriteConstants';
 WriteMakefile(
     NAME => 'GDBM_File',
-    LIBS => ["-lgdbm", "-ldbm"],
+    LIBS => ["-lgdbm -lgdbm_compat", "-ldbm"],
     XSPROTOARG => '-noprototypes', 		# XXX remove later?
     VERSION_FROM => 'GDBM_File.pm',
     realclean => {FILES=> 'const-c.inc const-xs.inc'},
