--- work/alien-classad-2.1.0/src/Makefile.am	Fri Mar 25 21:02:47 2005
+++ work/alien-classad-2.1.0/src/Makefile.am.new	Thu Sep  1 14:30:03 2005
@@ -12,7 +12,7 @@
 Classad.C \
 Classad_wrap.cxx
 
-libClassad_la_LIBADD = -L$(SWIGDIR)/lib -lswigpl -L$(CLASSADSDIR)/lib -l@CLASSADSLIB@  -L/usr/lib -lstdc++
+libClassad_la_LIBADD = -L$(CLASSADSDIR)/lib -l@CLASSADSLIB@  -L/usr/lib -lstdc++
 
 libClassad_la_LDFLAGS = -module -avoid-version
 
--- work/alien-classad-2.1.0/src/Makefile.am	Thu Sep  1 14:31:36 2005
+++ work/alien-classad-2.1.0/src/Makefile.am.new	Tue Sep  6 13:34:36 2005
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = foreign
 
-INCLUDES = -I. -I$(CLASSADSDIR)/include/classads -I$(CLASSADSDIR)/include -I$(SWIGDIR)/lib/swig1.3 -I$(SWIGDIR)/lib/swig1.3/perl5  -I${PERL5EXT}
+INCLUDES = -I. -I$(CLASSADSDIR)/include/classads -I$(CLASSADSDIR)/include -I${PERL5EXT}
 
 CXXFLAGS = @CXXFLAGS@ -DCLASSAD_DISTRIBUTION
 
