diff --speed-large-files --minimal -Nru tmp/Crypt-SSLeay-0.64/Makefile.PL work/Crypt-SSLeay-0.64/Makefile.PL
--- tmp/Crypt-SSLeay-0.64/Makefile.PL	2012-08-04 16:11:32.000000000 +0200
+++ work/Crypt-SSLeay-0.64/Makefile.PL	2013-02-05 15:17:00.000000000 +0100
@@ -67,7 +67,7 @@
 code is working properly. If you are behind a strict firewall or have no
 network connectivity, these tests may fail (through no fault of the code).
 EO_CHUNK
-    my $wanted = prompt "Do you want to run the live tests (y/N)?", 'N';
+    my $wanted = 'N';
     $wanted =~ s/\A\s+//;
     $wanted =~ s/\s+\z//;
 
