--- ./work/bbftp-client-3.2.0/bbftpc/bbftp_utils.c.ori	2005-10-20 15:20:33.000000000 +0200
+++ ./work/bbftp-client-3.2.0/bbftpc/bbftp_utils.c	2005-10-20 15:21:04.000000000 +0200
@@ -40,6 +40,16 @@
 #include <client.h>
 #include <client_proto.h>
 #include <netinet/in.h>
+#if TIME_WITH_SYS_TIME
+# include <sys/time.h>
+# include <time.h>
+#else
+# if HAVE_SYS_TIME_H
+#  include <sys/time.h>
+# else
+#  include <time.h>
+# endif
+#endif
 #include <structures.h>
 #include <config.h>
 #ifdef HAVE_BYTESWAP_H
