--- work/monit-4.5.1/process/sysdep_DARWIN.c.ori	2005-09-02 18:09:35.000000000 +0200
+++ work/monit-4.5.1/process/sysdep_DARWIN.c	2005-09-02 18:46:44.000000000 +0200
@@ -33,6 +33,7 @@
 #endif
 
 #ifdef HAVE_SYS_VMMETER_H
+#include <sys/types.h>
 #include <sys/vmmeter.h>
 #endif
 
@@ -170,7 +171,11 @@
     pt[i].cputime   = (long)((pinfo[i].kp_proc.p_rtime.tv_sec * 10) +
                              (pinfo[i].kp_proc.p_rtime.tv_usec / 100000)
                             );
+#ifdef never
     pt[i].mem_kbyte = (unsigned long)(pinfo[i].kp_eproc.e_vm.vm_rssize * pagesize_kbyte);
+#else
+    pt[i].mem_kbyte = (unsigned long)0;
+#endif
     if(pinfo[i].kp_proc.p_stat == SZOMB)
     {
       pt[i].status_flag |= PROCESS_ZOMBIE;
