--- ./work/alien-admin-2.1.1/src/etc/MonaLisa/AliEn/Monitoring.sh.ori	2005-09-29 11:11:01.000000000 +0200
+++ ./work/alien-admin-2.1.1/src/etc/MonaLisa/AliEn/Monitoring.sh	2005-09-29 11:35:11.000000000 +0200
@@ -47,7 +47,7 @@
 ###########################################################################
 Monitoring_CE() 
 {
-	name=`hostname -f`;
+	name=`hostname`;
 	info=`$1/bin/soapcall http://$name:8084 /AliEn/Service/ClusterMonitor getQueueInfo $2`;
 	oldremember="1";
 	remember="1";
--- ./work/alien-admin-2.1.1/src/etc/rc.d/init.d/alien-httpd.ori	2005-09-29 11:17:15.000000000 +0200
+++ ./work/alien-admin-2.1.1/src/etc/rc.d/init.d/alien-httpd	2005-09-29 11:25:11.000000000 +0200
@@ -56,7 +56,7 @@
 export ALIEN_PERL=$ALIEN_ROOT/bin/alien-perl
 
 export ALIEN_HOME=${ALIEN_HOME:=$HOME/.alien}
-domain=`hostname -d`
+domain=`hostname | sed -e "s/$(hostname -s)\.//"`
 
 export ALIEN_DOMAIN=${ALIEN_DOMAIN:=$domain}
 
--- ./work/alien-admin-2.1.1/src/scripts/CreateOrgDB.pl.ori	2005-09-29 11:11:01.000000000 +0200
+++ ./work/alien-admin-2.1.1/src/scripts/CreateOrgDB.pl	2005-09-29 11:35:01.000000000 +0200
@@ -40,7 +40,7 @@
 
 }
 
-my $hostName=`hostname -f`;
+my $hostName=`hostname`;
 chomp $hostName;
 $hostName=getParam("Host name", $hostName);
 
--- ./work/alien-admin-2.1.1/src/scripts/CreateOrgLDAP.pl.ori	2005-09-29 11:17:15.000000000 +0200
+++ ./work/alien-admin-2.1.1/src/scripts/CreateOrgLDAP.pl	2005-09-29 11:24:15.000000000 +0200
@@ -33,7 +33,7 @@
 print "Please, enter the following information:\n";
 my $orgName       = getParam("Organisation name","ALICE");
 
-my $suffix=`hostname -d`;
+my $suffix=`hostname | sed -e "s/$(hostname -s)\.//"`;
 chomp $suffix;
 $suffix=~ s/\./,dc=/g;
 $suffix="dc=$suffix";
@@ -51,7 +51,7 @@
 
 }
 
-my $hostName=`hostname -f`;
+my $hostName=`hostname`;
 chomp $hostName;
 $hostName=getParam("Host name", $hostName);
 
@@ -61,7 +61,7 @@
 
 my $mysqlHost=getParam("alien-mysql host",getDefaultMysql($orgName,$hostName));
 my $siteName=getParam("Site name", "CERN");
-my $domainName=`hostname -d`;
+my $domainName=`hostname | sed -e "s/$(hostname -s)\.//"`;
 chomp $domainName;
 $domainName=getParam("Domain of $siteName", $domainName);
 my $authenSubject=getParam("Certificate subject", getCertificateSubject());
--- ./work/alien-admin-2.1.1/src/scripts/CreateOrgPortal.pl.ori	2005-09-29 11:11:01.000000000 +0200
+++ ./work/alien-admin-2.1.1/src/scripts/CreateOrgPortal.pl	2005-09-29 11:34:50.000000000 +0200
@@ -12,7 +12,7 @@
 print "Please, enter the following information:\n";
 my $orgName       = getParam("Organisation name","ALICE");
 #my $alienRoot     =  getParam("AliEn Root directory", "$ENV{ALIEN_ROOT}");
-my $hostName=`hostname -f`;
+my $hostName=`hostname`;
 chomp $hostName;
 
 my $mysqlHost=getParam("alien-mysql host",getDefaultMysql($orgName,$hostName));
--- ./work/alien-admin-2.1.1/src/scripts/CreateOrgServices.pl.ori	2005-09-29 11:11:01.000000000 +0200
+++ ./work/alien-admin-2.1.1/src/scripts/CreateOrgServices.pl	2005-09-29 11:34:45.000000000 +0200
@@ -32,7 +32,7 @@
 
 $config or exit(-2);
 
-my $hostname=`hostname -f`;
+my $hostname=`hostname`;
 chomp $hostname;
 my $names={};
 $names->{AUTH}="Authen";
