--- MonaLisa.v1.6/Service/CMD/ML_SER	2007-07-16 10:08:24.000000000 +0200
+++ MonaLisa.v1.6/Service/CMD/ML_SER.new	2007-07-16 10:08:24.000000000 +0200
@@ -39,6 +39,7 @@
 PRGDIR=`dirname "$PRG"`
 CONFDIR=${CONFDIR:=$PRGDIR}
 COMMON_SH="${PRGDIR}/common.sh"
+ID_CMD="env LD_LIBRARY_PATH= id"
 if [ ! -z "$2" ]; then
  if [ "$2" == "from_su" ]; then
    FROM_SU="$2"
@@ -75,11 +76,11 @@
 
 if [ -z "${MONALISA_USER}" ]; then
      echo "Please set MONALISA_USER in ${ML_ENV_FILE}"
-     echo "e.g MONALISA_USER=`id -u -n`"
+     echo "e.g MONALISA_USER=`$ID_CMD -u -n`"
      exit 1
 fi
 
-cuser=`id -n -u`
+cuser=`$ID_CMD -n -u`
 if [ "${cuser}" != "${MONALISA_USER}" ]; then
         if [ "$FROM_SU" != "from_su" ]; then
           echo "The current user ( ${cuser} ) != MONALISA_USER ( ${MONALISA_USER} ) (as set in ${ML_ENV_FILE} ) will try to su in ${MONALISA_USER}."
@@ -283,7 +284,7 @@
 start() {
 	common
 
-	c_uid="`id -u -r`"
+	c_uid="`$ID_CMD -u -r`"
 	pid_cmd=`get_pid_cmd "$cmd_RE_update" $c_uid ""`
 	pupdate_pid=`$MLSER_SHELL -c "$pid_cmd"`
 	if [ ! -z "${pupdate_pid}" ]; then
@@ -311,7 +312,8 @@
 	   exit 1
 	fi
     fi
-    
+    # put the script's PID in the file until ML starts
+    echo $$ > ${ML_PID_FILE}
     if [ "${SHOULD_UPDATE}" = "true" ]; then 
         ml_update
     fi 
@@ -366,7 +368,7 @@
 checkedStopEDB() {
         common
 
-        c_uid="`id -u -r`"
+        c_uid="`$ID_CMD -u -r`"
         pid_cmd=`get_pid_cmd "$cmd_RE" $c_uid ""`
         pupdate_pid=`$MLSER_SHELL -c "$pid_cmd"`
         if [ -z "${pupdate_pid}" ]; then
@@ -400,7 +402,7 @@
 }
 
 stop_mljvm() {
-	c_uid="`id -u -r`"
+	c_uid="`$ID_CMD -u -r`"
 	ppid_cmd=`get_pid_cmd "${cmd_RE}" $c_uid 1`
 	pid_cmd=`get_pid_cmd "${cmd_RE}" $c_uid ""`
 	ppid=`$MLSER_SHELL -c "$ppid_cmd"`
@@ -417,7 +419,7 @@
     common; 
     rm -f "${ML_PID_FILE}"
 	
-	c_uid="`id -u -r`"
+	c_uid="`$ID_CMD -u -r`"
 	pupdate_pid_cmd=`get_pid_cmd "$cmd_RE_update" $c_uid ""`
 	pupdate_pid=`$MLSER_SHELL -c "$pupdate_pid_cmd"`
 	if [ ! -z "${pupdate_pid}" ]; then
@@ -550,7 +552,7 @@
 
 update(){
     common
-    c_uid="`id -u -r`"
+    c_uid="`$ID_CMD -u -r`"
     pupdate_pid_cmd=`get_pid_cmd "$cmd_RE_update" $c_uid ""`
     pupdate_pid=`$MLSER_SHELL -c "${pupdate_pid_cmd}"`
                                                                                                                                                
