--- work/glite-data-io-client-1.3.1/src/utils/Makefile.am	Wed Feb  2 17:00:09 2005
+++ work/glite-data-io-client-1.3.1/src/utils/Makefile.am.new	Fri Feb  4 15:26:33 2005
@@ -16,7 +16,7 @@
                glite-rm
 
 glite_data_io_client_test_ldflags   = \
-	-L$(top_srcdir)/build/src -lglite_data_io_client \
+	-L$(top_builddir)/src -lglite_data_io_client \
 	-L$(GLITE_LOCATION)/lib -lglite_data_config -lglite_data_io_base \
 	$(GLOBUS_GSS_NOTHR_LIBS) \
 	$(LOG4CPP_LIBS)
