--- work/stunnel-4.07/Makefile.am	2004-10-10 07:37:15.000000000 +0200
+++ new/stunnel-4.07/Makefile.am	2005-01-16 12:05:16.154215808 +0100
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = src doc tools
+SUBDIRS = src doc
 
 # extra_src = src/gui.c src/resources.rc src/stunnel.ico src/stunnel.exe
 # extra_doc = doc/stunnel.pod doc/stunnel.8 doc/stunnel.html doc/en doc/pl
