--- work/bbftp-3.0.1/bbftpc/configure.in	Mon Aug 11 23:11:41 2003
+++ work/bbftp-3.0.1/bbftpc/configure.in.new	Thu Jan 27 15:28:31 2005
@@ -622,7 +622,9 @@
         CCIN2P3_CHECK_RFIO([RFIOOBJS="bbftp_retr_rfio.o bbftp_store_rfio.o"],[AC_MSG_WARN(RFIO not installed)],[$with_rfio])
     fi
 fi
-
+if test "$WITH_RFIO64" != "no" ; then
+   CFLAGS="$CFLAGS -DWITH_RFIO64 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE"
+fi
 dnl ===========================================================================
 dnl Include OpenSSL as encrypting library if needed
 dnl Openssl is not necessary as it is provided with Globus
