--- work/e2fsprogs-1.39/Makefile.in	2006-11-06 12:23:00.000000000 +0100
+++ work/e2fsprogs-1.39/Makefile.in.new	2006-11-06 12:22:41.000000000 +0100
@@ -10,11 +10,11 @@
 @RESIZER_CMT@RESIZE_DIR= resize
 @DEBUGFS_CMT@DEBUGFS_DIR= debugfs
 
-LIB_SUBDIRS=lib/et lib/ss lib/e2p lib/ext2fs lib/uuid lib/blkid intl
-PROG_SUBDIRS=e2fsck $(DEBUGFS_DIR) misc $(RESIZE_DIR) tests/progs po
+LIB_SUBDIRS=lib/uuid
+PROG_SUBDIRS=
 SUBDIRS=util $(LIB_SUBDIRS) $(PROG_SUBDIRS) tests
 
-SUBS= lib/ext2fs/ext2_types.h lib/blkid/blkid_types.h lib/uuid/uuid_types.h
+SUBS= lib/uuid/uuid_types.h
 
 TAR=tar
 
