--- work/classads-0.9.7/util.C	Tue Feb 22 18:04:48 2005
+++ work/classads-0.9.7/util.C.new	Tue Feb 22 18:05:01 2005
@@ -27,8 +27,6 @@
 
 using namespace std;
 
-extern DLL_IMPORT_MAGIC long timezone;
-
 BEGIN_NAMESPACE( classad )
 
 #ifdef WIN32
--- ./work/classads-0.9.7/util.C	2006-08-18 08:53:53.000000000 +0200
+++ ./work/classads-0.9.7/util.C.ori	2006-08-18 08:51:57.000000000 +0200
@@ -353,16 +353,9 @@
 }
 #endif 
 
-#ifdef  __APPLE_CC__
-int classad_isnan(double x)
-{
-    return __isnan(x);
-}
-#else
 int classad_isnan(double x)
 {
     return isnan(x);
 }
-#endif
 
 END_NAMESPACE // classad
