diff --speed-large-files --minimal -Nru tmp/xrootd-xalienfs-1.0.14r1/configure work/xrootd-xalienfs-1.0.14r1/configure
--- tmp/xrootd-xalienfs-1.0.14r1/configure	2014-05-12 11:30:55.000000000 +0200
+++ work/xrootd-xalienfs-1.0.14r1/configure	2015-11-19 17:51:03.000000000 +0100
@@ -17089,8 +17089,8 @@
 $as_echo "$as_me: WARNING: xrd application will be built without GNU readline support" >&2;}
 fi
 
-CPPFLAGS="$CPPFLAGS_BUP"
-LDFLAGS="$LDFLAGS_BUP"
+#CPPFLAGS="$CPPFLAGS_BUP"
+#LDFLAGS="$LDFLAGS_BUP"
 SHELL_LIBS="$READLINE_LIBS"
 
 
