--- work/bbftp-3.0.1/bbftpd/Makefile.in	Wed Jun 11 17:18:39 2003
+++ new/bbftp-3.1.0/bbftpd/Makefile.in	Sun Jan 16 13:52:02 2005
@@ -81,8 +81,9 @@
 	@INSTALL@ ${PROG} ${bindir}/${PROG}
 	mkdir -p ${mandir}
 	mkdir -p ${mandir}/man1
+	mkdir -p ${prefix}/etc/rc.d/init.d/
 	@INSTALL@ ../doc/${PROG}.1 ${mandir}/man1/${PROG}.1
-	@INSTALL@ startup /etc/rc.d/init.d/${PROG}
+	@INSTALL@ startup ${prefix}/etc/rc.d/init.d/${PROG}
 
 runtest: ${PROG}
 	./${PROG} -v
