--- work/xrootd-1.0.0/xrootd/src/XrdOfs/GNUmakefile.castor	Tue Apr 26 18:01:33 2005
+++ work/xrootd-1.0.0/xrootd/src/XrdOfs/GNUmakefile.castor.new	Thu May 12 10:49:48 2005
@@ -20,7 +20,9 @@
         $(OBJDIR)/XrdOfsConfig.o  \
         $(OBJDIR)/XrdOfsHandle.o  
 
-MORELIBS = -L$(LIBDIR) -lXrdAcc -lXrdOss -lXrdOdc -lXrdXr -lXrdNet -lXrdOuc -lshift
+SHIFTLIB = $(wildcard /usr/lib/libshift.a /usr/local/lib/libshift.a)
+
+MORELIBS = -L$(LIBDIR) -lXrdAcc -lXrdOss -lXrdOdc -lXrdXr -lXrdNet -lXrdOuc $(SHIFTLIB)
 
 LIBDEP   = $(LIBDIR)/libXrdAcc.a $(LIBDIR)/libXrdOss.a \
            $(LIBDIR)/libXrdOdc.a $(LIBDIR)/libXrdOuc.a \
