--- work/globus-2.4.3ng/globus_common_setup-2.2/setup-globus-common.pl	Tue Sep 24 23:33:54 2002
+++ work/globus-2.4.3ng/globus_common_setup-2.2/setup-globus-common.pl.new	Mon Aug 29 17:55:11 2005
@@ -36,13 +36,13 @@
 
 for my $setupfile ('globus-script-initializer', 'globus-sh-tools-vars.sh')
 {
-    $result = system("cp $setupdir/$setupfile $globusdir/libexec");
+    $result = system("mv $setupfile $globusdir/libexec");
     $result = system("chmod 0755 $globusdir/libexec/$setupfile");
 
 }
 
 system("mkdir -p $globusdir/lib/perl/Globus/Core");
-system("cp $setupdir/Paths.pm $globusdir/lib/perl/Globus/Core/");
+system("mv Paths.pm $globusdir/lib/perl/Globus/Core/");
 system("chmod 0644 $globusdir/lib/perl/Globus/Core/Paths.pm");
 
 print "checking globus-hostname\n";
