--- ./work/root-5.12.0/config/Makefile.macosx.ori	2006-08-18 16:34:09.000000000 +0200
+++ ./work/root-5.12.0/config/Makefile.macosx	2006-08-18 16:34:30.000000000 +0200
@@ -80,11 +80,11 @@
 ifeq (g95,$(findstring g95,$(ROOTBUILD)))
 F77           = g95
 F77LIBS       = -L`$(F77) -print-search-dirs | awk '/^install:/{print $$2}'` \
-                -lf95 -L$(FINK_DIR)/lib -lg2c
+                -lf95 -L$(FINK_DIR)/lib
 else
 F77           = gfortran
 F77LIBS       = -L$(FINK_DIR)/lib/gcc4/lib -lgfortran -lgfortranbegin \
-                -L$(FINK_DIR)/lib -lg2c
+                -L$(FINK_DIR)/lib
 endif
 else
 F77           = g77
