--- work/root-5.13.4/mysql/inc/TMySQLServer.h	2006-07-17 10:42:39.000000000 +0200
+++ work/root-5.13.4/mysql/inc/TMySQLServer.h.new	2006-11-09 11:01:38.000000000 +0100
@@ -20,6 +20,7 @@
 #ifndef R__WIN32
 #include <sys/time.h>
 #endif
+#include <sys/types.h>
 #include <mysql.h>
 #else
 struct MYSQL;
--- work/root-5.13.4/mysql/inc/TMySQLStatement.h	2006-10-11 16:27:33.000000000 +0200
+++ work/root-5.13.4/mysql/inc/TMySQLStatement.h.new	2006-11-09 11:02:06.000000000 +0100
@@ -20,6 +20,7 @@
 #ifndef R__WIN32
 #include <sys/time.h>
 #endif
+#include <sys/types.h>
 #include <mysql.h>
 
 #if MYSQL_VERSION_ID < 40100
