--- ./work/lfc-1.3.7/LFC.spec.ori	2005-09-30 15:49:41.000000000 +0200
+++ ./work/lfc-1.3.7/LFC.spec	2005-09-30 15:49:56.000000000 +0200
@@ -29,7 +29,7 @@
 Summary: Client side libraries for the LFC
 Group: grid/lcg
 Requires: lcg-dm-common = @VERSION@
-Provides: liblfc.so
+Provides: libalfc.so
 Obsoletes: CSEC
 %description client
 APIs and CLIs for the LFC
@@ -127,9 +127,9 @@
 %attr(644,root,root) %{prefix}/include/lfc/lfc_api.h
 %attr(644,root,root) %{prefix}/include/lfc/osdep.h
 %attr(644,root,root) %{prefix}/include/lfc/serrno.h
-%attr(755,root,root) %{prefix}/lib/liblfc.a
-%attr(755,root,root) %{prefix}/lib/liblfc.so
-%attr(755,root,root) %{prefix}/lib/liblfc.so.%{version}
+%attr(755,root,root) %{prefix}/lib/libalfc.a
+%attr(755,root,root) %{prefix}/lib/libalfc.so
+%attr(755,root,root) %{prefix}/lib/libalfc.so.%{version}
 %attr(644,root,root) %{prefix}/share/doc/%{name}-client-%{version}/README
 %attr(644,root,root) %{prefix}/share/doc/%{name}-client-%{version}/INSTALL-client
 %attr(755,root,root) %{prefix}/bin/lfc-chmod
--- ./work/lfc-1.3.7/lib/Imakefile.ori	2005-09-30 16:08:17.000000000 +0200
+++ ./work/lfc-1.3.7/lib/Imakefile	2005-09-30 16:08:51.000000000 +0200
@@ -18,14 +18,14 @@
 #endif
 
 #if BuildNameServerAsLFC
-COMM Libraries to include in LibraryTargetName(lfc)
+COMM Libraries to include in LibraryTargetName(alfc)
 LIBRARIES = $(COMMONLIB) $(NSLIB) $(SECURITYLIB)
 
-all:    LibraryTargetName(lfc)
+all:    LibraryTargetName(alfc)
 
-install: $(LIB) FileName($(LIB),LibraryTargetName(lfc))
+install: $(LIB) FileName($(LIB),LibraryTargetName(alfc))
 
-export: $(EXPORTLIB)/liblfc.a
+export: $(EXPORTLIB)/libalfc.a
 #else
 #if BuildNameServerAsDPNS
 COMM Libraries to include in LibraryTargetName(dpm)
@@ -43,7 +43,7 @@
 
 exportshr: 
 
-libdpm.a liblfc.a:     $(LIBRARIES)
+libdpm.a libalfc.a:     $(LIBRARIES)
 	@echo " making $@ in `pwd`"
 	@-rm -rf tmp
 	@mkdir tmp
@@ -60,7 +60,7 @@
 MakeDir($(LIB),root,bin,0755)
 
 InstallLibrary(dpm,$(LIB),root,bin,644)
-InstallLibrary(lfc,$(LIB),root,bin,644)
+InstallLibrary(alfc,$(LIB),root,bin,644)
 
 install.man:
 
@@ -69,7 +69,7 @@
 	-mv $(EXPORTLIB)/$?  $(EXPORTLIB)/OLD$?
 	cp $? $(EXPORTLIB)/$?
 	@$(RANLIB) $(EXPORTLIB)/$?
-$(EXPORTLIB)/liblfc.a: liblfc.a
+$(EXPORTLIB)/libalfc.a: libalfc.a
 	@(if [ ! -d $(EXPORTLIB) ] ; then mkdir -p $(EXPORTLIB) ; fi ; )
 	-mv $(EXPORTLIB)/$?  $(EXPORTLIB)/OLD$?
 	cp $? $(EXPORTLIB)/$?
--- ./work/lfc-1.3.7/shlib/Imakefile.ori	2005-09-30 16:05:51.000000000 +0200
+++ ./work/lfc-1.3.7/shlib/Imakefile	2005-09-30 16:06:34.000000000 +0200
@@ -19,14 +19,14 @@
 #endif
 
 #if BuildNameServerAsLFC
-COMM Libraries to include in SharedLibraryTargetName(lfc)
+COMM Libraries to include in SharedLibraryTargetName(alfc)
 LIBRARIES = $(COMMONLIB) $(NSLIB) $(SECURITYLIB)
 
-all:    SharedLibraryTargetName(lfc)
+all:    SharedLibraryTargetName(alfc)
 
-install: $(LIB) FileName($(LIB),SharedLibraryTargetName(lfc))
+install: $(LIB) FileName($(LIB),SharedLibraryTargetName(alfc))
 
-export: $(EXPORTLIB)/SharedLibraryTargetName(lfc)
+export: $(EXPORTLIB)/SharedLibraryTargetName(alfc)
 #else
 #if BuildNameServerAsDPNS
 COMM Libraries to include in SharedLibraryTargetName(dpm)
@@ -44,7 +44,7 @@
 
 exportshr: 
 
-libdpm.sl libdpm.so liblfc.sl liblfc.so libdpm.dylib liblfc.dylib:     $(LIBRARIES)
+libdpm.sl libdpm.so libalfc.sl libalfc.so libdpm.dylib libalfc.dylib:     $(LIBRARIES)
 	@echo " making $@ in `pwd`"
 	@-rm -rf tmp
 	@mkdir tmp
@@ -60,10 +60,10 @@
 MakeDir($(LIB),root,bin,0755)
 
 InstallSharedLibrary(dpm,$(VERSION),$(LIB))
-InstallSharedLibrary(lfc,$(VERSION),$(LIB))
+InstallSharedLibrary(alfc,$(VERSION),$(LIB))
 
 InstallSharedLibrary(dpm,$(VERSION),$(EXPORTLIB))
-InstallSharedLibrary(lfc,$(VERSION),$(EXPORTLIB))
+InstallSharedLibrary(alfc,$(VERSION),$(EXPORTLIB))
 
 install.man:
 
