--- work/lfc-1.3.7/common/Cuuid.c.ori	2005-08-30 11:39:21.000000000 +0200
+++ work/lfc-1.3.7/common/Cuuid.c	2005-08-30 11:41:43.000000000 +0200
@@ -36,7 +36,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
-#ifndef _WIN32
+#if (!defined(_WIN32) && !defined(__APPLE__))
 #include <sys/sysinfo.h>
 #endif
 #if (defined(aix) || defined(hpux))
@@ -775,7 +775,9 @@
 {
     MD5_CTX c;
     typedef struct {
+#ifndef __APPLE__
         struct sysinfo s;
+#endif
         struct timeval t;
         char hostname[257];
     } randomness;
@@ -806,8 +808,10 @@
 	}
 #endif
 #else /* aix || hpux */
+#ifndef __APPLE__
     sysinfo(&r.s);
 #endif
+#endif
     gettimeofday(&r.t, (struct timezone *)0);
     gethostname(r.hostname, 256);
     _Cuuid_MD5Update(&c, (unsigned char *) &r, sizeof(randomness));
--- work/lfc-1.3.7/dli/Imakefile.ori	2005-08-31 14:36:50.000000000 +0200
+++ work/lfc-1.3.7/dli/Imakefile	2005-08-31 14:37:04.000000000 +0200
@@ -46,9 +46,9 @@
 	$(CC) -c $(GSOAP)/stdsoap2.c
 
 dliH.h dliStub.h dliC.c dliClient.c dliServer.c DataLocationInterface.nsmap: DataLocationInterface.h
-	$(GSOAP)/soapcpp2 -c -p dli DataLocationInterface.h
+	soapcpp2 -c -p dli DataLocationInterface.h
 DataLocationInterface.h: DataLocationInterface.wsdl
-	$(GSOAP)/wsdl2h -c DataLocationInterface.wsdl
+	wsdl2h -c DataLocationInterface.wsdl
 
 NormalProgramTarget(lfc-dli,$(SRV_OBJS), $(DEPLIB), $(MTLDFLAGS) $(LIBS) $(SECLIB) $(MTLDLIBS))
 NormalProgramTarget(dli_testlistReplicas,dli_testlistReplicas.Osuf dliC.Osuf dliClient.Osuf stdsoap2.Osuf,$(DEPLIB),$(SECLIB))
--- work/lfc-1.3.7/test/srmv1/Imakefile.ori	2005-08-31 14:36:29.000000000 +0200
+++ work/lfc-1.3.7/test/srmv1/Imakefile	2005-08-31 14:36:41.000000000 +0200
@@ -52,9 +52,9 @@
 	$(CC) -c $(GSOAP)/stdsoap2.c
 
 srmv1H.h srmv1Stub.h srmv1C.c srmv1Client.c ISRM.nsmap: srm.v1.1.h
-	$(GSOAP)/soapcpp2 -c -p srmv1 srm.v1.1.h
+	soapcpp2 -c -p srmv1 srm.v1.1.h
 srm.v1.1.h: srm.v1.1.wsdl
-	$(GSOAP)/wsdl2h -c -t typemap.dat srm.v1.1.wsdl
+	wsdl2h -c -t typemap.dat srm.v1.1.wsdl
 
 NormalProgramTarget(srm_testadvisoryDelete,srm_testadvisoryDelete.Osuf $(SOAP_OBJS),$(DEPLIBS),$(LIBS) $(SECLIB))
 NormalProgramTarget(srm_testget,srm_testget.Osuf $(SOAP_OBJS),$(DEPLIBS),$(LIBS) $(SECLIB))
--- work/lfc-1.3.7/test/srmv2/Imakefile.ori	2005-08-31 14:35:57.000000000 +0200
+++ work/lfc-1.3.7/test/srmv2/Imakefile	2005-08-31 14:35:49.000000000 +0200
@@ -75,9 +75,9 @@
 	$(CC) -c $(GSOAP)/stdsoap2.c
 
 srmv2H.h srmv2Stub.h srmv2C.c srmv2Client.c srmSoapBinding.nsmap: srm.v2.1.1.h
-	$(GSOAP)/soapcpp2 -c -p srmv2 srm.v2.1.1.h
+	soapcpp2 -c -p srmv2 srm.v2.1.1.h
 srm.v2.1.1.h: srm.v2.1.1.wsdl
-	$(GSOAP)/wsdl2h -c -t typemap.dat srm.v2.1.1.wsdl
+	wsdl2h -c -t typemap.dat srm.v2.1.1.wsdl
 
 NormalProgramTarget(srm2_testGet,srm2_testGet.Osuf $(SOAP_OBJS),$(DEPLIBS),$(LIBS) $(SECLIB))
 NormalProgramTarget(srm2_testPut,srm2_testPut.Osuf $(SOAP_OBJS),$(DEPLIBS),$(LIBS) $(SECLIB))
--- work/lfc-1.3.7/dli/Imakefile.ori	2005-08-31 14:41:08.000000000 +0200
+++ work/lfc-1.3.7/dli/Imakefile	2005-08-31 14:52:20.000000000 +0200
@@ -15,7 +15,7 @@
 GLOBUS_LIBS=-L$(GLOBUS_LOCATION)/lib -lglobus_gssapi_gsi_$(GLOBUS_FLAVOUR) -lglobus_gss_assist_$(GLOBUS_FLAVOUR)
 GSOAP_LOCATION=GsoapLocation
 GSOAP_VERSION=GsoapVersion
-GSOAP = FileName($(GSOAP_LOCATION),gsoap-$(GSOAP_PLATFORM)-$(GSOAP_VERSION))
+GSOAP = $(GSOAP_LOCATION)/share/soapcpp2
 SECLIB = -lcgsi_plugin_gsoap_$(GSOAP_VERSION) $(GLOBUS_LIBS)
 INCLUDES = FileName(..,h)
 LIBS = -L../lib -llfc -ldl
--- work/lfc-1.3.7/test/srmv1/Imakefile.ori	2005-08-31 14:49:07.000000000 +0200
+++ work/lfc-1.3.7/test/srmv1/Imakefile	2005-08-31 14:51:00.000000000 +0200
@@ -16,7 +16,7 @@
 GLOBUS_LIBS=-L$(GLOBUS_LOCATION)/lib -lglobus_gssapi_gsi_$(GLOBUS_FLAVOUR) -lglobus_gss_assist_$(GLOBUS_FLAVOUR)
 GSOAP_LOCATION=GsoapLocation
 GSOAP_VERSION=GsoapVersion
-GSOAP = FileName($(GSOAP_LOCATION),gsoap-$(GSOAP_PLATFORM)-$(GSOAP_VERSION))
+GSOAP = $(GSOAP_LOCATION)/share/soapcpp2
 SECFLG = -DGFAL_SECURE
 SECLIB = -lcgsi_plugin_gsoap_$(GSOAP_VERSION) $(GLOBUS_LIBS)
 RFIOFLG = -DGFAL_ENABLE_RFIO
--- work/lfc-1.3.7/test/srmv2/Imakefile.ori	2005-08-31 14:47:24.000000000 +0200
+++ work/lfc-1.3.7/test/srmv2/Imakefile	2005-08-31 14:50:36.000000000 +0200
@@ -16,7 +16,7 @@
 GLOBUS_LIBS=-L$(GLOBUS_LOCATION)/lib -lglobus_gssapi_gsi_$(GLOBUS_FLAVOUR) -lglobus_gss_assist_$(GLOBUS_FLAVOUR)
 GSOAP_LOCATION=GsoapLocation
 GSOAP_VERSION=GsoapVersion
-GSOAP = FileName($(GSOAP_LOCATION),gsoap-$(GSOAP_PLATFORM)-$(GSOAP_VERSION))
+GSOAP = $(GSOAP_LOCATION)/share/soapcpp2
 SECFLG = -DGFAL_SECURE
 SECLIB = -lcgsi_plugin_gsoap_$(GSOAP_VERSION) $(GLOBUS_LIBS)
 RFIOFLG = -DGFAL_ENABLE_RFIO
--- ./work/lfc-1.3.7/scripts/choose-module.ori	2005-09-24 21:15:59.000000000 +0200
+++ ./work/lfc-1.3.7/scripts/choose-module	2005-09-24 21:16:41.000000000 +0200
@@ -37,7 +37,7 @@
 	`perl -ibak -npe "s#BuildSRMv1Server\tYES#BuildSRMv1Server\tNO#g" ../config/site.def`
 	`perl -ibak -npe "s#BuildSRMv2Server\tYES#BuildSRMv2Server\tNO#g" ../config/site.def`
         `perl -ibak -npe "s#BuildTest\t\tYES#BuildTest\t\tNO#g" ../config/site.def`
-	`perl -ibak -npe "s#BuildRfioClient\t\tYES#BuildRfioClient\t\tNO#g" ../config/site.def`
+#	`perl -ibak -npe "s#BuildRfioClient\t\tYES#BuildRfioClient\t\tNO#g" ../config/site.def`
 	`perl -ibak -npe "s#BuildRfioServer\t\tYES#BuildRfioServer\t\tNO#g" ../config/site.def`
         `perl -ibak -npe "s#UseGSI\t\tNO#UseGSI\t\tYES#g" ../config/site.def`
         `perl -ibak -npe "s#UseKRB5\t\tYES#UseKRB5\t\tNO#g" ../config/site.def`
--- ./work/lfc-1.3.7/shlib/Imakefile.ori	2005-09-30 15:29:53.000000000 +0200
+++ ./work/lfc-1.3.7/shlib/Imakefile	2005-09-30 15:30:08.000000000 +0200
@@ -50,7 +50,7 @@
 	@mkdir tmp
 	@for i in $(LIBRARIES) ;\
 	do (cd tmp; $(AR) x ../$$i; rm -f $(SYMTABNAM)) done
-	@(cd tmp; ld $(SHLIBLDFLAGS) -o ../$@ *.o $(SHLIBREQLIBS) $(LIBCSEC))
+	@(cd tmp; g++ $(SHLIBLDFLAGS) -o ../$@ *.o $(SHLIBREQLIBS) $(LIBCSEC))
 	@-rm -rf tmp
 
 dpm.dll lfc.dll:	$(LIBRARIES)
--- ./work/lfc-1.3.7/config/darwin.cf.ori	2005-10-09 11:48:29.000000000 +0200
+++ ./work/lfc-1.3.7/config/darwin.cf	2005-10-09 11:48:42.000000000 +0200
@@ -11,7 +11,7 @@
 MTCCFLAGS = -DCTHREAD_POSIX -D_THREAD_SAFE -D_REENTRANT
 MTLDLIBS =
 MTLDFLAGS =
-SHLIBLDFLAGS = -undefined dynamic_lookup -dylib -single_module
+SHLIBLDFLAGS = -undefined dynamic_lookup -dynamiclib -single_module
 SHLIBREQLIBS = -lc
 FC = g77
 
--- work/lfc-1.3.7/imake/imakemdep.h.ori  2006-04-11 18:03:14.000000000 +0200
+++ work/lfc-1.3.7/imake/imakemdep.h	2005-02-14 11:56:45.000000000 +0100
@@ -148,7 +148,7 @@
 #ifdef __ia64__
 #define DEFAULT_CPP "/usr/bin/cpp"
 #endif
-#if defined( __ppc__) && defined( __APPLE__)
+#if (defined( __ppc__) || defined( __i386__)) && defined( __APPLE__)
 #define DEFAULT_CPP "/usr/bin/cpp"
 #endif
 
--- work/lfc-1.3.7/config/Imake.tmpl	2005-02-14 11:36:57.000000000 +0100
+++ work/lfc-1.3.7/config/Imake.tmpl.new	2006-04-12 10:22:20.000000000 +0200
@@ -16,6 +16,9 @@
 #if __ppc__ && __APPLE__
 #include <darwin.cf>
 #endif
+#if __i386__ && __APPLE__
+#include <darwin.cf>
+#endif
 #if hpux
 #include <hp.cf>
 #endif
--- work/lfc-1.3.7/imake/imakemdep.h	2006-04-11 18:51:06.000000000 +0200
+++ work/lfc-1.3.7/imake/imakemdep.h.new	2006-04-12 10:21:30.000000000 +0200
@@ -103,6 +103,10 @@
 #define imake_ccflags "-DSYSV"
 #endif
 
+#if defined( __i386__) && defined( __APPLE__)
+#define imake_ccflags "-DSYSV"
+#endif
+
 #else /* not CCIMAKE */
 #ifndef MAKEDEPEND
 /*
