--- work/ejabberd-1.1.1/src/Makefile.in	2006-04-11 15:48:24.000000000 +0200
+++ work.new2/ejabberd-1.1.1/src/Makefile.in	2006-04-11 15:32:44.000000000 +0200
@@ -1,7 +1,7 @@
 # $Id$
 
 CC = @CC@
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS@ -fno-common -bundle -undefined suppress -flat_namespace
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
--- work.ori/ejabberd-1.1.1/src/mod_irc/Makefile.in	2005-05-25 15:31:59.000000000 +0200
+++ work/ejabberd-1.1.1/src/mod_irc/Makefile.in	2006-04-11 15:34:39.000000000 +0200
@@ -1,7 +1,7 @@
 # $Id$
 
 CC = @CC@
-CFLAGS = @CFLAGS@ @ERLANG_CFLAGS@
+CFLAGS = @CFLAGS@ @ERLANG_CFLAGS@ -fno-common -bundle -undefined suppress -flat_namespace
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@ @ERLANG_LIBS@ @LIBICONV@
--- work.ori/ejabberd-1.1.1/src/stringprep/Makefile.in	2005-05-25 15:31:59.000000000 +0200
+++ work/ejabberd-1.1.1/src/stringprep/Makefile.in	2006-04-11 15:36:07.000000000 +0200
@@ -1,7 +1,7 @@
 # $Id$
 
 CC = @CC@
-CFLAGS = @CFLAGS@ @ERLANG_CFLAGS@
+CFLAGS = @CFLAGS@ @ERLANG_CFLAGS@ -fno-common -bundle -undefined suppress -flat_namespace
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@ @ERLANG_LIBS@
--- work.ori/ejabberd-1.1.1/src/tls/Makefile.in	2005-11-19 18:25:36.000000000 +0100
+++ work/ejabberd-1.1.1/src/tls/Makefile.in	2006-04-11 15:36:24.000000000 +0200
@@ -1,7 +1,7 @@
 # $Id$
 
 CC = @CC@
-CFLAGS = @CFLAGS@ @SSL_CFLAGS@ @ERLANG_CFLAGS@
+CFLAGS = @CFLAGS@ @SSL_CFLAGS@ @ERLANG_CFLAGS@ -fno-common -bundle -undefined suppress -flat_namespace
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@ @SSL_LIBS@ @ERLANG_LIBS@
