--- work/Authen-SASL-2.10/lib/Authen/SASL.pm	2005-04-26 15:34:23.000000000 +0200
+++ work/Authen-SASL-2.10/lib/Authen/SASL.pm.new	2006-08-09 15:07:49.000000000 +0200
@@ -71,7 +71,8 @@
 
   foreach my $pkg (@Plugins) {
     if (eval "require $pkg" and $pkg->can("client_new")) {
-      return ($self->{conn} = $pkg->client_new($self, @_));
+      $self->{conn} = eval { $pkg->client_new($self, @_); };
+      return $self->{conn} if $self->{conn};
     }
   }
 
