--- ./work/glite-data-1.2.9/bin/cvs2cl.ori	2005-09-29 11:17:15.000000000 +0200
+++ ./work/glite-data-1.2.9/bin/cvs2cl	2005-09-29 11:21:05.000000000 +0200
@@ -2067,7 +2067,7 @@
         chomp($Domain = slurp_file(MAILNAME));
       } else {
       MAILDOMAIN_CMD:
-        for ([qw(hostname -d)], 'dnsdomainname', 'domainname') {
+        for ([qw(hostname | sed -e "s/$(hostname -s)\.//")], 'dnsdomainname', 'domainname') {
           my ($text, $exit, $sig, $core) = run_ext($_);
           if ( $exit == 0 && $sig == 0 && $core == 0 ) {
             chomp $text;
