--- work/xrootd-1.0.7/xrootd/src/XrdSfs/XrdSfsAio.hh.ori	2005-08-25 14:58:15.000000000 +0200
+++ work/xrootd-1.0.7/xrootd/src/XrdSfs/XrdSfsAio.hh	2005-08-25 14:58:50.000000000 +0200
@@ -57,11 +57,7 @@
 virtual void Recycle() = 0;
 
              XrdSfsAio() {
-#ifdef __macos__
-                         sfsAio.aio_sigevent.sigev_value.sigval_ptr = (void *)this;
-#else
                          sfsAio.aio_sigevent.sigev_value.sival_ptr  = (void *)this;
-#endif
                          sfsAio.aio_sigevent.sigev_notify = SIGEV_SIGNAL;
                          sfsAio.aio_reqprio = 0;
                          TIdent = (char *)"";
--- work/xrootd-1.0.7/configure.ac.ori	2005-06-03 12:20:15.000000000 +0200
+++ work/xrootd-1.0.7/configure.ac	2005-08-25 15:02:22.000000000 +0200
@@ -4,6 +4,7 @@
 AC_CONFIG_SRCDIR([xrootd/config/GNUmakefile.in])
 AM_INIT_AUTOMAKE([xrootd], 1.0.0)
 AM_CONFIG_HEADER(config.h)
+AC_CANONICAL_HOST()
 
 # Checks for programs.
 AC_PROG_CXX
@@ -37,6 +38,16 @@
 AC_SUBST(XPLATFORM)
 
 
+case "$host_os" in
+    darwin*)
+        SOEXT="dylib"
+    ;;
+    *)
+        SOEXT="so"
+    ;;
+esac
+ 
+AC_SUBST(SOEXT)
 
 AC_CONFIG_FILES([Makefile])
 
--- work/xrootd-1.0.7/Makefile.am.ori	2005-08-25 15:07:35.000000000 +0200
+++ work/xrootd-1.0.7/Makefile.am	2005-08-25 15:02:42.000000000 +0200
@@ -7,11 +7,11 @@
 	xrootd/bin/arch/xrdcp\
 	xrootd/bin/arch/xrm\
 	xrootd/bin/arch/xrootd\
-	xrootd/lib/arch/libXrdEnvOfs.so\
-	xrootd/lib/arch/libXrdOfs.so\
-	xrootd/lib/arch/libXrdPosix.so\
-	xrootd/lib/arch/libXrdRootd.so\
-	xrootd/lib/arch/libXrdSec.so
+	xrootd/lib/arch/libXrdEnvOfs.$(SOEXT)\
+	xrootd/lib/arch/libXrdOfs.$(SOEXT)\
+	xrootd/lib/arch/libXrdPosix.$(SOEXT)\
+	xrootd/lib/arch/libXrdRootd.$(SOEXT)\
+	xrootd/lib/arch/libXrdSec.$(SOEXT)
 
 SCRIPT_FILES = testing/ardautils/castor_meta.pl \
 	testing/ardautils/castor_migrate.pl \
@@ -23,11 +23,11 @@
 	xrootd/bin/arch/xrm\
 	xrootd/bin/arch/xrootd
 
-LIB_FILES = xrootd/lib/arch/libXrdEnvOfs.so\
-	xrootd/lib/arch/libXrdOfs.so\
-	xrootd/lib/arch/libXrdPosix.so\
-	xrootd/lib/arch/libXrdRootd.so\
-	xrootd/lib/arch/libXrdSec.so
+LIB_FILES = xrootd/lib/arch/libXrdEnvOfs.$(SOEXT)\
+	xrootd/lib/arch/libXrdOfs.$(SOEXT)\
+	xrootd/lib/arch/libXrdPosix.$(SOEXT)\
+	xrootd/lib/arch/libXrdRootd.$(SOEXT)\
+	xrootd/lib/arch/libXrdSec.$(SOEXT)
 
 INC_FILES = `cd xrootd/src; find . -name "*.hh"`
 
--- work/xrootd-1.0.7/xrootd/config/GNUmake.rules.macos.ori	2005-08-25 15:08:03.000000000 +0200
+++ work/xrootd-1.0.7/xrootd/config/GNUmake.rules.macos	2005-08-25 15:03:31.000000000 +0200
@@ -15,8 +15,8 @@
 TYPELINK   = -undefined dynamic_lookup -multiply_defined suppress 
 TYPEOPT    = $(TYPEMISC) $(TYPELINK)
 TYPEDBG    = $(TYPEMISC) -g
-TYPELDSO   = $(TYPELINK) -bundle
-TYPESHLIB  = so
+TYPELDSO   = $(TYPELINK) -dynamiclib
+TYPESHLIB  = dylib
 
 # In MacOSx the threading functions should come libc.
 #TYPELIBS   =  $(PTHREAD) -ldl -lc
--- work/xrootd-1.0.7/xrootd/src/XrdEnvOfs/XrdEnvOfs.cc.ori	2005-08-25 15:08:33.000000000 +0200
+++ work/xrootd-1.0.7/xrootd/src/XrdEnvOfs/XrdEnvOfs.cc	2005-08-25 15:04:13.000000000 +0200
@@ -9,6 +9,10 @@
 #include <sys/time.h>
 #include <stdarg.h>
 
+#ifdef __macos__
+#define EUNATCH 49
+#endif
+
 // if you want to allow the alien shell to do normal work define ALIEN_BACKDOOR
 #define ALIEN_BACKDOOR
 
--- work/xrootd-1.0.7/xrootd/src/XrdEnvOfs/GNUmakefile.ori	2005-08-25 15:15:33.000000000 +0200
+++ work/xrootd-1.0.7/xrootd/src/XrdEnvOfs/GNUmakefile	2005-08-25 15:16:05.000000000 +0200
@@ -16,7 +16,7 @@
 OBJECTS  = \
         $(OBJDIR)/XrdEnvOfs.o 
 
-MORELIBS = -L$(EXTINSTALL)/lib/ -L$(LIBDIR) -lXrdAcc -lXrdOss -lXrdOdc -lXrdXr -lXrdNet -lXrdOuc -lXrdOfs -lAuthzXMLreader -lSealedEnvelope 
+MORELIBS = -L$(EXTINSTALL)/lib/ -L$(LIBDIR) -lXrdAcc -lXrdOss -lXrdOdc -lXrdXr -lXrdNet -lXrdOuc $(LIBDIR)/libXrdOfs.a -lAuthzXMLreader -lSealedEnvelope 
 
 LIBDEP   = 
 
--- work/xrootd-1.0.7/xrootd/src/XrdOfs/GNUmakefile.ori	2005-08-25 15:11:59.000000000 +0200
+++ work/xrootd-1.0.7/xrootd/src/XrdOfs/GNUmakefile	2005-08-25 15:14:50.000000000 +0200
@@ -28,9 +28,10 @@
            $(LIBDIR)/libXrdOdc.a $(LIBDIR)/libXrdOuc.a \
            $(LIBDIR)/libXrdXr.a		
 
+LIBARCH  = $(LIBDIR)/libXrdOfs.a
 LIBRARY  = $(LIBDIR)/libXrdOfs.$(TYPESHLIB)
 
-TARGETS  = $(LIBRARY)
+TARGETS  = $(LIBARCH) $(LIBRARY)
 
 EXTINC   = -I $(EXTINSTALL)/include
 
@@ -60,7 +61,18 @@
 #                           D e p e n d e n c i e s                            #
 #------------------------------------------------------------------------------#
 
-$(TARGETS): $(OBJECTS) $(LIBDEP)
+$(LIBARCH): $(OBJECTS)
+	@echo Creating archive $(LIBARCH) 
+	$(ECHO)rm -f $(LIBARCH)
+	$(ECHO)ar -rc $(LIBARCH) $(OBJECTS)
+	$(ECHO)ranlib $(LIBARCH)
+	$(ECHO)if [ "x$(SUNCACHE)" != "x" ]; then \
+	   if [ "$(TYPE)" = "SunOS" -a "$(CC)" = "CC" ]; then \
+	      ar -rc $(LIBARCH) $(OBJDIR)$(SUNCACHE)/*/*.o; \
+	   fi;\
+	fi
+
+$(LIBRARY): $(OBJECTS) $(LIBDEP)
 	@echo Creating shared library $(LIBRARY) 
 	$(ECHO)$(CC) $(CFLAGS) $(OBJECTS) $(LDSO) $(MORELIBS) $(LIBS)  -o $(LIBRARY)
 
--- work/xrootd-1.0.7/xrootd/configure.ori	2005-08-25 15:31:00.000000000 +0200
+++ work/xrootd-1.0.7/xrootd/configure	2005-08-25 15:31:27.000000000 +0200
@@ -334,7 +334,7 @@
     if test ! "x$symboldir" = "x" ; then
         symbolfile=$symboldir/$symbollib
         if test ! -r $symbolfile ; then
-            for i in .a .so .lib ; do
+            for i in .a .so .lib .dylib ; do
                 if test -r $symbolfile$i ; then
                     symbolfile=$symbolfile$i
                     break
--- work/xrootd-1.0.7/xrootd/src/XrdClient/GNUmakefile.ori	2005-08-25 15:32:05.000000000 +0200
+++ work/xrootd-1.0.7/xrootd/src/XrdClient/GNUmakefile	2005-08-25 15:33:01.000000000 +0200
@@ -80,7 +80,11 @@
 XRM = $(BINDIR)/xrm
 PERLMOD =
 ifneq ($(PERLINC),)
-PERLMOD = $(LIBDIR)/XrdClientAdmin.so
+ifneq ($(shell uname -s),Darwin)
+ PERLMOD = $(LIBDIR)/XrdClientAdmin.so
+else
+PERLMOD = $(LIBDIR)/XrdClientAdmin.dylib
+endif
 endif
 
 
--- work/xrootd-1.0.7/xrootd/src/XrdSec/XrdSecPManager.cc.ori	2005-08-25 15:33:38.000000000 +0200
+++ work/xrootd-1.0.7/xrootd/src/XrdSec/XrdSecPManager.cc	2005-08-25 15:34:20.000000000 +0200
@@ -212,7 +212,11 @@
 
 // Form library name
 //
-   snprintf(libfn, sizeof(libfn)-1, "libXrdSec%s.so", pid);
+#ifdef __macos__
+   snprintf(libfn, sizeof(libfn)-1, "libXrdSec%s.dylib", pid);
+#else
+    snprintf(libfn, sizeof(libfn)-1, "libXrdSec%s.so", pid);
+#endif
    libfn[sizeof(libfn)-1] = '\0';
 
 // Determine path
