--- work/glite-security-voms-admin-client-1.0.7/Makefile.PL	Mon Sep 12 17:40:41 2005
+++ work/glite-security-voms-admin-client-1.0.7/Makefile.PL.new	Mon Sep 12 17:40:21 2005
@@ -0,0 +1,13 @@
+use ExtUtils::MakeMaker;
+use Config;
+
+
+# See lib/ExtUtils/MakeMaker.pm for details of how to influence
+# the contents of the Makefile that is written.
+WriteMakefile(
+          'NAME'               => 'gLite::VOMS',
+          'PREREQ_PM'       => {
+              'SOAP::Lite'   => 0,
+          },
+          'EXE_FILES'         => ['scripts/voms-admin']
+);
