--- ./work/cgsi-gsoap-1.1.5/Makefile.in.ori	2005-02-21 19:18:46.000000000 +0100
+++ ./work/cgsi-gsoap-1.1.5/Makefile.in	2005-09-28 08:44:01.000000000 +0200
@@ -11,7 +11,7 @@
 CFLAGS = -Wall -g -fPIC
 LDFLAGS =
 RANLIB = ranlib
-SHLIBLDFLAGS = -shared
+SHLIBLDFLAGS = -dynamiclib $(OWN_LDFLAGS)
 SHLIBREQLIBS = -lc 
 
 ################################################################################
@@ -30,7 +30,7 @@
 ################################################################################
 ## compilation targets ##
 .PHONY: all
-all: libcgsi_plugin$(GSOAP_VERSION).a libcgsi_plugin$(GSOAP_VERSION).so 
+all: libcgsi_plugin$(GSOAP_VERSION).a libcgsi_plugin$(GSOAP_VERSION).$(GARLIBEXT)
 
 # dependencies
 cgsi_plugin.o: cgsi_plugin.h cgsi_plugin_int.h
@@ -39,8 +39,8 @@
 	$(AR) $(ARFLAGS) $@ $?
 	$(RANLIB) $@
 
-libcgsi_plugin$(GSOAP_VERSION).so: cgsi_plugin.o
-	ld $(SHLIBLDFLAGS) -o $@ *.o $(SHLIBREQLIBS)
+libcgsi_plugin$(GSOAP_VERSION).$(GARLIBEXT): cgsi_plugin.o
+	g++ $(SHLIBLDFLAGS) -o $@ *.o $(SHLIBREQLIBS)
 
 ###############################################
 ## THIS IS STANDARD Makefile BELOW THIS LINE ##
@@ -60,7 +60,7 @@
 # insert space separated files to install (used for cleaning too) here
 BINFILES =
 LIBFILES =libcgsi_plugin$(GSOAP_VERSION).a
-SHLIBFILES =libcgsi_plugin$(GSOAP_VERSION).so
+SHLIBFILES =libcgsi_plugin$(GSOAP_VERSION).$(GARLIBEXT)
 INCFILES =cgsi_plugin.h
 MANFILES =
 
