*** work/Term-ReadLine-Gnu-1.20/Makefile.PL.old	2012-05-09 07:07:40.000000000 +0200
--- work/Term-ReadLine-Gnu-1.20/Makefile.PL	2012-05-09 07:07:56.000000000 +0200
*************** if ($Config{osname} eq 'os2') {
*** 60,65 ****
--- 60,66 ----
      # Old Cygwin may require setting false (0).
      my $PREFER_CURSES = $Config{osname} eq 'aix' || $Config{osname} eq 'hpux'
  			|| $Config{osname} eq 'cygwin';
+     $PREFER_CURSES =1;
      my $TERMCAP_LIB = (! $PREFER_CURSES && &search_lib('-ltermcap'))
  	|| &search_lib('-lncurses')
  	    || &search_lib('-lcurses');
