--- ./work/alien-common-2.2.5/src/scripts/Alien.sh.ori	2005-09-23 16:22:41.000000000 +0200
+++ ./work/alien-common-2.2.5/src/scripts/Alien.sh	2005-09-23 16:22:55.000000000 +0200
@@ -596,7 +596,11 @@
      do
        if [ -d $dir ]
        then 
-         file=`find $dir -path $1 -xtype f -print -maxdepth 3 2>/dev/null`  
+         if test "`uname -s`" = "Darwin" ; then 
+           file=`find -L $dir -path $1 -type f -print -maxdepth 3 2>/dev/null`  
+         else
+           file=`find $dir -path $1 -xtype f -print -maxdepth 3 2>/dev/null`  
+	 fi
          if [ -f "$file" ]
          then
            bindir=`dirname $file`
@@ -628,6 +632,8 @@
   ALIEN_ORGANISATION=""
   ALIEN_USER=""
 
+  if test "`uname -s`" = "Darwin" ; then DY_EXT=dylib ; else DY_EXT=so ; fi
+
   config_scope="user"
 
   while [ $# -gt 0 ]
@@ -760,9 +766,9 @@
   # cgsi-gsoap
   #########
 
-  CGSI_GSOAP_LOCATION=`FindLocation "*/lib/libcgsi_plugin_gsoap_2.6.so" $CGSI_GSOAP_LOCATION`
+  CGSI_GSOAP_LOCATION=`FindLocation "*/lib/libcgsi_plugin_gsoap_2.6.${DY_EXT}" $CGSI_GSOAP_LOCATION`
 
-  if [ -f $CGSI_GSOAP_LOCATION/lib/libcgsi_plugin_gsoap_2.6.so ]
+  if [ -f $CGSI_GSOAP_LOCATION/lib/libcgsi_plugin_gsoap_2.6.${DY_EXT} ]
   then
     ALIEN_LD_LIBRARY_PATH=$CGSI_GSOAP_LOCATION/lib:$ALIEN_LD_LIBRARY_PATH
   else
@@ -774,9 +780,9 @@
   # classad
   #########
 
-  CLASSAD_LOCATION=`FindLocation "*/lib/libaclassad.so" $CLASSAD_LOCATION`
+  CLASSAD_LOCATION=`FindLocation "*/lib/libaclassad.${DY_EXT}" $CLASSAD_LOCATION`
 
-  if [ -f $CLASSAD_LOCATION/lib/libaclassad.so ]
+  if [ -f $CLASSAD_LOCATION/lib/libaclassad.${DY_EXT} ]
   then
     ALIEN_LD_LIBRARY_PATH=$CLASSAD_LOCATION/lib:$ALIEN_LD_LIBRARY_PATH
   else
