--- ./work/ejabberd-0.9.8/src/Makefile.in.ori	2005-09-20 16:53:41.000000000 +0200
+++ ./work/ejabberd-0.9.8/src/Makefile.in	2005-09-20 16:41:46.000000000 +0200
@@ -15,7 +15,7 @@
 prefix = @prefix@
 
 SUBDIRS = @mod_irc@ @mod_pubsub@ @mod_muc@ @eldap@ @web@ stringprep @tls@ @odbc@
-ERLSHLIBS = expat_erl.so
+ERLSHLIBS = expat_erl.dylib
 
 DESTDIR =
 
@@ -43,12 +43,14 @@
 	done
 
 
-$(ERLSHLIBS):	%.so:	%.c
-			gcc -Wall $(CFLAGS) $(LDFLAGS) $(LIBS) \
-			$(subst ../,,$(subst .so,.c,$@)) \
+$(ERLSHLIBS):	%.dylib:	%.c
+			MACOSX_DEPLOYMENT_TARGET=10.4 $(CC) \
+			-dynamiclib -single_module -undefined dynamic_lookup \
+			-Wall $(CFLAGS) $(LDFLAGS) $(LIBS) \
+			$(subst ../,,$(subst .dylib,.c,$@)) \
 			$(EXPAT_LIBS) $(EXPAT_CFLAGS) \
 			$(ERLANG_LIBS) $(ERLANG_CFLAGS) \
-			-o $@ -fpic -shared
+			-o $@ 
 
 install: all
 	install -d $(BEAMDIR)
@@ -56,7 +58,7 @@
 	rm -f $(BEAMDIR)/configure.beam
 	install -m 644 *.app $(BEAMDIR)
 	install -d $(SODIR)
-	install -m 644 *.so $(SODIR)
+	install -m 644 *.dylib $(SODIR)
 	install -d $(MSGSDIR)
 	install -m 644 msgs/*.msg $(MSGSDIR)
 	install -d $(ETCDIR)
--- ./work/ejabberd-0.9.8/src/mod_irc/Makefile.in.ori	2005-09-20 16:53:41.000000000 +0200
+++ ./work/ejabberd-0.9.8/src/mod_irc/Makefile.in	2005-09-20 16:41:32.000000000 +0200
@@ -8,7 +8,7 @@
 
 SUBDIRS = 
 
-ERLSHLIBS = ../iconv_erl.so
+ERLSHLIBS = ../iconv_erl.dylib
 
 OUTDIR = ..
 EFLAGS = -I .. -pz ..
@@ -25,10 +25,12 @@
 #all:	$(ERLSHLIBS)
 #	erl -s make all report "{outdir, \"..\"}" -noinput -s erlang halt
 
-$(ERLSHLIBS):	../%.so:	%.c
-			$(CC) -Wall $(INCLUDES) $(CFLAGS) $(LDFLAGS) \
-			$(subst ../,,$(subst .so,.c,$@)) $(LIBS) \
-			-o $@ -fpic -shared
+$(ERLSHLIBS):	../%.dylib:	%.c
+			MACOSX_DEPLOYMENT_TARGET=10.4 $(CC) \
+			-dynamiclib -single_module -undefined dynamic_lookup \
+			-Wall $(INCLUDES) $(CFLAGS) $(LDFLAGS) \
+			$(subst ../,,$(subst .dylib,.c,$@)) $(LIBS) \
+			-o $@
 
 clean:
 	rm -f $(OBJS) $(ERLSHLIBS)
--- ./work/ejabberd-0.9.8/src/stringprep/Makefile.in.ori	2005-09-20 16:53:41.000000000 +0200
+++ ./work/ejabberd-0.9.8/src/stringprep/Makefile.in	2005-09-20 16:41:19.000000000 +0200
@@ -8,7 +8,7 @@
 
 SUBDIRS = 
 
-ERLSHLIBS = ../stringprep_drv.so
+ERLSHLIBS = ../stringprep_drv.dylib
 
 OUTDIR = ..
 EFLAGS = -I .. -pz ..
@@ -23,10 +23,12 @@
 #all:	$(ERLSHLIBS)
 #	erl -s make all report "{outdir, \"..\"}" -noinput -s erlang halt
 
-$(ERLSHLIBS):	../%.so:	%.c uni_data.c uni_norm.c
-			gcc -Wall -O2 $(CFLAGS) $(LDFLAGS) $(INCLUDES) \
-			$(subst ../,,$(subst .so,.c,$@)) $(LIBS) \
-			-o $@ -fpic -shared
+$(ERLSHLIBS):	../%.dylib:	%.c uni_data.c uni_norm.c
+			MACOSX_DEPLOYMENT_TARGET=10.4 $(CC) \
+			-dynamiclib -single_module -undefined dynamic_lookup \
+			-Wall -O2 $(CFLAGS) $(LDFLAGS) $(INCLUDES) \
+			$(subst ../,,$(subst .dylib,.c,$@)) $(LIBS) \
+			-o $@ 
 
 clean:
 	rm -f $(OBJS) $(ERLSHLIBS)
--- ./work/ejabberd-0.9.8/src/tls/Makefile.in.ori	2005-09-20 16:53:41.000000000 +0200
+++ ./work/ejabberd-0.9.8/src/tls/Makefile.in	2005-09-20 16:40:56.000000000 +0200
@@ -8,7 +8,7 @@
 
 SUBDIRS = 
 
-ERLSHLIBS = ../tls_drv.so
+ERLSHLIBS = ../tls_drv.dylib
 
 OUTDIR = ..
 EFLAGS = -I .. -pz ..
@@ -23,10 +23,12 @@
 #all:	$(ERLSHLIBS)
 #	erl -s make all report "{outdir, \"..\"}" -noinput -s erlang halt
 
-$(ERLSHLIBS):	../%.so:	%.c
-			$(CC) -Wall $(CFLAGS) $(LDFLAGS) \
-			$(subst ../,,$(subst .so,.c,$@)) $(LIBS) \
-			-o $@ -fpic -shared
+$(ERLSHLIBS):	../%.dylib:	%.c
+			MACOSX_DEPLOYMENT_TARGET=10.4 $(CC) \
+			-dynamiclib -snigle_module -undefined dynamic_lookup \
+			-Wall $(CFLAGS) $(LDFLAGS) \
+			$(subst ../,,$(subst .dylib,.c,$@)) $(LIBS) \
+			-o $@
 
 clean:
 	rm -f $(OBJS) $(ERLSHLIBS)
