--- work/xrootd-xrdcpapmonplugin-1.1.0/configure.old	2010-07-12 15:36:09.000000000 +0200
+++ work/xrootd-xrdcpapmonplugin-1.1.0/configure	2010-07-12 15:36:27.000000000 +0200
@@ -723,13 +723,6 @@
 MAKEFLAGS=
 SHELL=${CONFIG_SHELL-/bin/sh}
 
-# Identity of this package.
-PACKAGE_NAME='xrootd-xrdcpapmonplugin'
-PACKAGE_TARNAME='xrootd-xrdcpapmonplugin'
-PACKAGE_VERSION='1.1.0'
-PACKAGE_STRING='xrootd-xrdcpapmonplugin 1.1.0'
-PACKAGE_BUGREPORT='xrootd-l@slac.stanford.edu'
-
 ac_unique_file="XrdCpapMon.cc"
 # Factoring default headers for most tests.
 ac_includes_default="\
