--- work/libgpg-error-1.4/configure	2006-03-14 13:50:56.000000000 +0100
+++ work/libgpg-error-1.4/configure.new	2006-05-18 00:43:14.000000000 +0200
@@ -26182,8 +26182,8 @@
       if test "$nls_cv_use_gnu_gettext" = "yes"; then
                 BUILD_INCLUDED_LIBINTL=yes
         USE_INCLUDED_LIBINTL=yes
-        LIBINTL="\${top_builddir}/intl/libintl.a $LIBICONV"
-        LTLIBINTL="\${top_builddir}/intl/libintl.a $LTLIBICONV"
+        LIBINTL="\${top_builddir}/intl/libintl.\$la $LIBICONV"
+        LTLIBINTL="\${top_builddir}/intl/libintl.\$la $LTLIBICONV"
         LIBS=`echo " $LIBS " | sed -e 's/ -lintl / /' -e 's/^ //' -e 's/ $//'`
       fi
 
@@ -26309,7 +26309,7 @@
     fi
 
 
-        INTL_LIBTOOL_SUFFIX_PREFIX=
+        INTL_LIBTOOL_SUFFIX_PREFIX=l
 
 
 
--- work/libgpg-error-1.4/intl/relocatable.c	2006-03-14 13:21:16.000000000 +0100
+++ work/libgpg-error-1.4/intl/relocatable.c.new	2006-05-18 00:38:19.000000000 +0200
@@ -399,7 +399,7 @@
 	 in a different prefix before being moved to orig_prefix and (later)
 	 to curr_prefix, 2. unlike the program, it has not moved away from
 	 orig_prefix.  */
-      const char *orig_installprefix = INSTALLPREFIX;
+      const char *orig_installprefix = " "/*INSTALLPREFIX*/;
       const char *orig_installdir = INSTALLDIR;
       const char *curr_prefix_better;
 
--- work/libgpg-error-1.4/src/Makefile.in	2006-03-14 13:50:55.000000000 +0100
+++ work/libgpg-error-1.4/src/Makefile.in.new	2006-05-18 07:20:17.000000000 +0200
@@ -334,6 +334,8 @@
   @LIBGPG_ERROR_LT_CURRENT@:@LIBGPG_ERROR_LT_REVISION@:@LIBGPG_ERROR_LT_AGE@ \
   $(no_undefined) $(export_symbols)
 
+l = @INTL_LIBTOOL_SUFFIX_PREFIX@
+
 libgpg_error_la_SOURCES = gpg-error.h gettext.h $(arch_sources) \
 	init.c strsource.c strerror.c code-to-errno.c code-from-errno.c
 
