--- work/globus-2.4.3ng/globus_openssl-0.20/configure.in	Thu Nov 28 22:03:17 2002
+++ work/globus-2.4.3ng/globus_openssl-0.20/configure.in.new	Fri Oct  7 11:44:44 2005
@@ -57,8 +57,9 @@
 
 # Need to determine if we need -ldl or -ldld
 # maybe should be replaced with the libtool dl library
-CHECK_NEED_LDL
+# CHECK_NEED_LDL
 
+EXTERNAL_LIBS="$EXTERNAL_LIBS -ldl"
 
 GPT_SET_LIBS([[$EXTERNAL_LIBS]])	
 
