--- work/Data-UUID-0.11/UUID.xs	Wed Aug 27 21:38:35 2003
+++ work/Data-UUID-0.11/UUID.xs.new	Thu Jun  2 13:48:46 2005
@@ -2,6 +2,8 @@
 #include "perl.h"
 #include "XSUB.h"
 #include "UUID.h"
+#include <sys/types.h>
+#include <unistd.h>
 
 static  uuid_t NameSpace_DNS = { /* 6ba7b810-9dad-11d1-80b4-00c04fd430c8 */
    0x6ba7b810,
@@ -141,6 +143,7 @@
    } randomness;
 #else
    typedef struct {
+      pid_t          pid;
       long           hostid;
       struct timeval t;
       char           hostname[257];
@@ -159,6 +162,7 @@
    r.l = MAX_COMPUTERNAME_LENGTH + 1;
    GetComputerName(r.hostname, &r.l );
 #else
+   r.pid = getpid();
    r.hostid = gethostid();
    gettimeofday(&r.t, (struct timezone *)0);
    gethostname(r.hostname, 256);
