--- work/SealedEnvelope-1.0.2/TSealedEnvelope.cxx	Thu Aug 11 16:00:01 2005
+++ work/SealedEnvelope-1.0.2/TSealedEnvelope.cxx.new	Tue Aug 30 13:36:21 2005
@@ -1,7 +1,10 @@
+#ifdef __APPLE__
+#define uuid_t notype_t
+#endif
+
 #include "TSealedEnvelope.h"
 #include <iostream>
 #include <sstream>
-#include <uuid/uuid.h>
 #include <netinet/in.h>
 #include <unistd.h>
 #include <zlib.h>
@@ -11,6 +14,12 @@
 #include <fcntl.h>
 #include <map>
 
+#ifdef __APPLE__
+#undef uuid_t
+#endif
+
+#include <uuid.h>
+
 #ifdef WITHTHREADS
 #include <pthread.h>
 pthread_mutex_t lock;
