--- work/alien-api-1.0.5a/C2Perl/C2PERL.c.ori	2005-08-31 17:08:38.000000000 +0200
+++ work/alien-api-1.0.5a/C2Perl/C2PERL.c	2005-08-31 17:09:15.000000000 +0200
@@ -6,6 +6,8 @@
 #include <stdarg.h>
 #include <string.h>
 
+int g_init;
+
 static PerlInterpreter *my_perl;
 
 static void xs_init (void);
--- work/alien-api-1.0.5a/C2Perl/C2PERL.h.ori	2005-08-31 17:09:25.000000000 +0200
+++ work/alien-api-1.0.5a/C2Perl/C2PERL.h	2005-08-31 17:09:59.000000000 +0200
@@ -12,7 +12,7 @@
 #endif
     
     /* the initialization state of the perl module */
-    int g_init;
+    extern int g_init;
     
 /**************************************************************************
  * C2PERL API data types.
--- work/alien-api-1.0.5a/gclientlib/glite_dir_operations.cxx.ori	2005-08-31 17:21:47.000000000 +0200
+++ work/alien-api-1.0.5a/gclientlib/glite_dir_operations.cxx	2005-08-31 17:33:31.000000000 +0200
@@ -57,8 +57,10 @@
     delete dir;
     return 0;
   }
-  
+
+#ifndef __APPLE__  
   dir->dir.glite_dir.dir_entry.d_off=0;
+#endif
   int column=0;
   for(column=0; ;column++){
     std::map<std::string, std::string> tags;
@@ -91,16 +93,26 @@
   if(dir->type == sys)
     return readdir(dir->dir.sys_dir);
   struct dirent *dir_entry=&dir->dir.glite_dir.dir_entry;
+#ifndef __APPLE__
   int len = strspn(dir->dir.glite_dir.dirstream+dir_entry->d_off, "\n ");
   dir_entry->d_off+=len;
   len = strcspn(dir->dir.glite_dir.dirstream+dir_entry->d_off, "\n ");
+#else
+  int len = strspn(dir->dir.glite_dir.dirstream, "\n ");
+#endif
   if(!len) 
     return 0;
   if(len>255) len=255;
+#ifndef __APPLE__
   strncpy(dir->dir.glite_dir.dir_entry.d_name,
 	  dir->dir.glite_dir.dirstream +dir_entry->d_off, len);
   dir->dir.glite_dir.dir_entry.d_name[len]='\0';
   dir_entry->d_off += len;
+#else
+  strncpy(dir->dir.glite_dir.dir_entry.d_name,
+	  dir->dir.glite_dir.dirstream, len);
+  dir->dir.glite_dir.dir_entry.d_name[len]='\0';
+#endif
   return dir_entry;
 }
 
@@ -128,7 +140,11 @@
   if(dir->type == sys)
     return telldir(dir->dir.sys_dir);
 
+#ifndef __APPLE__
   return dir->dir.glite_dir.dir_entry.d_off;
+#else
+  return 0;
+#endif
 }
 
 
@@ -139,7 +155,9 @@
   if(dir->type == sys)
     rewinddir(dir->dir.sys_dir);
 
+#ifndef __APPLE__
   dir->dir.glite_dir.dir_entry.d_off = 0;
+#endif
 }
 
 
@@ -150,7 +168,9 @@
   if(dir->type == sys)
     seekdir(dir->dir.sys_dir, offset);
 
+#ifndef __APPLE__
   dir->dir.glite_dir.dir_entry.d_off = offset;
+#endif
 }
 
 
