--- work/alien-api-1.0.5a/clients/Makefile.am	Mon May  9 17:37:53 2005
+++ work/alien-api-1.0.5a/clients/Makefile.am.new	Tue Oct  4 11:28:24 2005
@@ -99,11 +99,11 @@
 	cd $(srcdir)/scripts; for name in $(GSCRIPTBIN); do cat gprefix_$$name | sed s/GPREFIX/$(GPREFIX)/g | sed s/GSHELL_VERSION/$(VERSION)/g > $(GPREFIX)_$$name ;done
 	cd $(srcdir)/scripts; for name in $(GSCRIPTETC); do cat gprefix_$$name | sed s/GPREFIX/$(GPREFIX)/g | sed s/GSHELL_VERSION/$(VERSION)/g > $(GPREFIX)_$$name ;done
 	for name in $(GSCRIPTBIN); do $(INSTALL) -m 755 $(srcdir)/scripts/$(GPREFIX)_$$name $(DESTDIR)/$(bindir) ; done
-	for name in $(GSCRIPTETC); do $(INSTALL) -m 644 $(srcdir)/scripts/$(GPREFIX)_$$name $(DESTDIR)/$(bindir)/../etc/ ; done
+	for name in $(GSCRIPTETC); do $(INSTALL) -m 644 $(srcdir)/scripts/$(GPREFIX)_$$name $(DESTDIR)/$(sysconfdir)/ ; done
 	cd $(DESTDIR)/$(bindir); ln -f -s gbbox g$(GPREFIX)   
 	cd $(DESTDIR)/$(bindir) ln -f -s gps $(GPREFIX)_ps
 	cd $(DESTDIR)/$(bindir); ln -f -s gsubmit $(GPREFIX)_submit
-	$(INSTALL) -m 755 $(srcdir)/scripts/env_gbbox.sh $(DESTDIR)/$(bindir)/../etc;
+	$(INSTALL) -m 755 $(srcdir)/scripts/env_gbbox.sh $(DESTDIR)/$(sysconfdir)/;
 
 install-globus-lib-local:
 	cp $(GLOBUS_LOCATION)/lib/libglobus_callout_$(GLOBUS_FLAVOR).so.0 $(DESTDIR)/lib/
