--- work/AuthzXMLreader-1.0.2/configure.ac	Tue Apr 26 15:23:58 2005
+++ work/AuthzXMLreader-1.0.2/configure.ac.new	Wed Jun 22 15:29:34 2005
@@ -26,7 +26,7 @@
 AC_CHECK_HEADERS([arpa/inet.h fcntl.h float.h limits.h netdb.h netinet/in.h stdlib.h string.h strings.h sys/socket.h sys/time.h sys/timeb.h unistd.h])
 #    check for readline dependencies for Shell feature
 AC_CHECK_HEADERS([readline/readline.h readline/history.h curses.h term.h])
-AC_CHECK_HEADERS([security/pam_appl.h],,AC_MSG_ERROR([no PAM library headers found - install the PAM development package]))
+AC_CHECK_HEADERS([pam/pam_appl.h],,AC_MSG_ERROR([no PAM library headers found - install the PAM development package]))
 
 # Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
