--- work/bbftp-client-3.2.0/bbftpc/Makefile.in	Tue May  3 15:53:56 2005
+++ work/bbftp-client-3.2.0/bbftpc/Makefile.in.new	Fri Oct 14 15:29:16 2005
@@ -65,7 +65,7 @@
 
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-GFWLIB = @GFWLIB@
+GFWLIB = -L@prefix@/lib -lgfw -lltdl 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 bindir = @bindir@
@@ -76,10 +76,7 @@
 ${OBJS} : ${HEAD}
 ${STATUS_OBJS} : ${HEAD}
 
-../gfw/libgfw.a:
-	(cd ../gfw; make clean; make)
-
-${PROG} :${OBJS} ${GFWLIB}
+${PROG} :${OBJS}
 	${CC} -o $@ ${OBJS} ${LDFLAGS} ${GFWLIB} ${LIBS} 
 
 ${STATUS_PROG} :${STATUS_OBJS}
