--- work/Filesys-DiskFree-0.06/DiskFree.pm_orig	2008-12-05 12:31:13.221572484 +0100
+++ work/Filesys-DiskFree-0.06/DiskFree.pm	2008-12-05 12:33:08.714420359 +0100
@@ -122,6 +122,25 @@
     close(HANDLE) or croak("Cannot df $!");
 }
 
+sub df_dir(){
+    my $self=shift;
+    my $cmd="df";
+
+    $cmd=$self->command() or
+       croak "No df command known for format ".$self->{'FORMAT'};
+    if(@_){
+       if($_[0]){
+           open(HANDLE,"$cmd $_[0]|") or croak("Cannot fork $!");
+        }
+    }
+    if($self->load(\*HANDLE)){
+       close(HANDLE);
+       return 'true';
+    }
+    close(HANDLE) or croak("Cannot df $!");
+}
+
 sub load()  {
     my $self=shift;
     my $handle=shift;
@@ -300,6 +319,13 @@
 
 Perfoms a 'df' command, and stores the values for later use.
 
+=item Filesys::DiskFree->df_dir($dir)
+
+Perfoms a 'df' command in the given directory $dir, and stores the
+values for later use. This command only has problems with NFS if
+$dir is not reachable, wheras df() hangs if any of the NFS mounted
+directories is not reachable. Tested only for Linux.
+
 =item Filesys::DiskFree->command()
 
 Returns the appropriate command to do a 'df' command, for the current
