--- work/classads-0.9.7/Makefile	Tue Feb 22 22:32:04 2005
+++ work/classads-0.9.7/Makefile.new	Tue Feb 22 22:41:28 2005
@@ -5,7 +5,9 @@
 
 LD_FLAGS = $(EXTRA_LD_FLAGS) -L. -l$(LIB_LINK_NAME)
 
-all:: $(LIB_NAME) libclassad.so
+LIBCLASSAD=libclassad.$(GARLIBEXT)
+
+all:: $(LIB_NAME) $(LIBCLASSAD)
 
 LIB_INSTALL = $(DESTDIR)/lib
 BIN_INSTALL = $(DESTDIR)/bin
@@ -32,9 +34,9 @@
 	rm -f  $(LIB_NAME)
 	ar vrs $(LIB_NAME)  $(LIB_OBJ)
 
-libclassad.so: $(LIB_OBJ)
-	rm -f libclassad.so
-	$(CPlusPlus) -shared $(LIB_OBJ) -o libclassad.so
+$(LIBCLASSAD): $(LIB_OBJ)
+	rm -f $(LIBCLASSAD)
+	$(CPlusPlus) $(GARSHARED) $(LD_FLAGS) $(LIB_OBJ) -o $(LIBCLASSAD)
 
 instantiations.o:   instantiations.C
 	$(CPlusPlus) $(INST_C_PLUS_FLAGS) -c instantiations.C -o instantiations.o
@@ -67,14 +69,14 @@
 	$(CPlusPlus) -o sample sample.o $(LD_FLAGS)
 
 shared: shared.C
-	$(CPlusPlus) -fPIC -shared -o libshared.so shared.C
+	$(CPlusPlus) -fPIC $(GARSHARED) -o libshared.so shared.C
 
 install:: $(LIB_NAME) classad_functional_tester classad_unit_tester classad_version
 	mkdir -p $(LIB_INSTALL)
 	mkdir -p $(BIN_INSTALL)
 	mkdir -p $(INCLUDE_INSTALL)
 	cp $(LIB_NAME)  $(LIB_INSTALL)
-	cp libclassad.so $(LIB_INSTALL)
+	cp $(LIBCLASSAD) $(LIB_INSTALL)
 	cp $(HEADER_FILES) $(INCLUDE_INSTALL)
 
 clean: tidy
