--- work/lcg-gfal-1.7.3/srm_ifce.c	2005-07-20 09:30:38.000000000 +0200
+++ work/lcg-gfal-1.7.3/srm_ifce.c.new	2005-09-27 11:47:44.089134312 +0200
@@ -245,7 +245,7 @@
 	return (n);
 }
 
-srm_turlsfromsurls (int nbfiles, const char **surls, LONG64 *filesizes, char **protocols, int oflag, int *reqid, 
+srm_turlsfromsurls (int nbfiles, const char **surls, GFAL_LONG64 *filesizes, char **protocols, int oflag, int *reqid, 
 		    int **fileids, char **token, char ***turls, char *errbuf, int errbufsz, int timeout)
 {
 	int *f;
