--- work/jabberd-2.0s9/configure.in	Sat Mar 26 15:32:27 2005
+++ work/jabberd-2.0s9/configure.in.new	Sat Mar 26 15:38:40 2005
@@ -130,11 +130,13 @@
 if test "x-$want_mysql" = "x-yes" ; then
     AC_CHECK_HEADERS(mysql.h)
     if test "x-$ac_cv_header_mysql_h" != "x-yes" ; then
-        for incpath in /usr/include/mysql /usr/local/include/mysql; do
+        for mpath in $prefix /usr/include /usr/local/include; do
             if test "x-$ac_cv_header_mysql_h" != "x-yes" ; then
-                AC_MSG_CHECKING([for mysql.h in $incpath])
+                AC_MSG_CHECKING([for mysql.h in $mpath/include/mysql])
                 save_cppflags="$CPPFLAGS"
-                CPPFLAGS="-I$incpath $CPPFLAGS"
+                save_ldflags="$LDFLAGS"
+                CPPFLAGS="-I$mpath/include/mysql $CPPFLAGS"
+                LDFLAGS="-L$mpath/lib $LDFLAGS"
                 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <mysql.h>]])],
                                   [AC_MSG_RESULT(yes)
                                    AC_DEFINE(HAVE_MYSQL_H,,[Define if you have mysql.h])
@@ -142,6 +144,7 @@
                                   AC_MSG_RESULT(no))
                 if test "x-$ac_cv_header_mysql_h" != "x-yes" ; then
                     CPPFLAGS="$save_cppflags"
+                    LDFLAGS="$save_ldflags"
                 fi
             fi
         done
